src/ferc_xbrl_extractor/datapackage.py

```diff
@@ -332,6 +334,36 @@ def get_period_type(self):
         period_type = "instant" if "date" in self.schema_.primary_key else "duration"
         return period_type
 
+    def merge_resources(self, other: "Resource", other_version: str) -> "Resource":
+        """Merge same resource from multiple taxonomies."""
+        if self.schema_.primary_key != other.schema_.primary_key:
+            raise RuntimeError(
+                f"Can't merge resource {self.name} when versions have incompatible schemas"
+            )
+        original_fields = {field.name for field in self.schema_.fields}
+        other_fields = {field.name for field in other.schema_.fields}
+
+        if missing_fields := original_fields - other_fields:
```

Comment:
To clarify: does this mean we'll lose the metadata for older columns that have been dropped in the newer data?

Reply:
Yeah the older columns will remain, they're just logged so we can see what changed. I've added this to the docstring now.